iset.mm

```diff
@@ -98005,6 +98021,113 @@ This definition (in terms of ` U. ` and ` ~<_ ` ) is not taken directly
     syl wceq sizeinf adantl breqtrrd ) ACDZEBFGZHAIJZKBIJZLUCUEKLGZUDUCUEMDUGUC
     UEANOUEPRQUDUFKSUCBTUAUB $.
 
+  $( A finite set is equinumerous to the range of integers from one up to the
+     hash value of the set.  (Contributed by Jim Kingdon, 22-Feb-2022.) $)
+  isfinite4im $p |- ( A e. Fin -> ( 1 ... ( size ` A ) ) ~~ A ) $=
+    ( cfn wcel c1 csize cfv cfz co wceq cen wbr cn0 sizecl sizefz1 syl wb cz 1z
+    nn0zd fzfig sylancr sizeen mpancom mpbid ) ABCZDAEFZGHZEFUFIZUGAJKZUEUFLCUH
+    AMZUFNOUGBCZUEUHUIPUEDQCUFQCUKRUEUFUJSDUFTUAUGAUBUCUD $.
+
+  $( Two ways of saying a finite set is empty.  (Contributed by Jim Kingdon,
+     22-Feb-2022.) $)
```

Comment:
I do not know if there are agreements about or conventions for the contributors of theorems transferred from set.mm to iset.mm? Shouldn't the contributor of the theorem in set.mm be taken (unless there is a significant change of the proof)? For ~size0, for example, the proof seems to be exactly the same.

Reply:
My practice has been to change the contributor  if I change the name and syntax but not to consider whether the proof is similar or not. It is an imperfect rule as you note but it does have the virtue of being easy to follow.

Reply:
@jkingdon - I think when in doubt we should give credit to previous contributors. In particular, if the proof is similar, also give credit to the original contributor. Having multiple contributors is fine!

Reply:
We don't usually add a Contributed line for renames, do we? If so, I should just remove myself from most of the theorems in this pull request (which is fine with me) and use the Contributed line from set.mm.

As for the more general point, I'm not too sure and if anyone wants to audit Contributed lines against git logs I'm almost surely quite willing to let someone else try to figure out where to draw the lines. We definitely do have some practices around only adding these comments for the biggest or most defining contributions but I don't know if I always pick exactly the best place to draw the line (the history of contributions is, of course, credited in more detail in git logs).

Reply:
Maybe add also: "A finite set has size zero if and only if it is empty."

Reply:
As for attribution/credits, they indeed deal with two separate issues: (1) giving credit to the right person, (2) inform about the history/versions of a statement, without "ethical" concern.  Therefore, things are a bit vague one (1) and (2) lead to different solutions. For the present statement, which is minor, it is not very important in my opinion, and I'll go with what @jkingdon decides.